### ISSUE TITLE
[BUG][GoCardless] Account disconnects every month

### ISSUE THREAD

**Issue by LucBerge** [PipedreamHQ/pipedream#13025]
**Describe the bug**
I have a GoCardless account setup and I am using it in the "new transaction" trigger.

**To Reproduce**

Prerequisits : Create requisition link

1. Create an empty script
2. Add the "New transaction" trigger
3. Connect the account
4. Test the workflow/trigger
5. Wait 1 month

**Expected behavior**
GoCardless account stay connected forever

**Current behavior**
The GoCardless account disconnects after one month

**Additional context**
It could link with #4894 : The token must be refreshed every month

**Comment by malexanderlim** [comment]
Hi @LucBerge , I've opened up a support ticket with GoCardless but they are unable to process it as I do not have an active production GoCardless account - can you email me at michael [at] pipedream.com with the email address associated with your GoCardless account so that they can help look into the issue?

**Comment by LucBerge** [comment]
Hello @malexanderlim , I won't send you my credentials. You would have access to my transactions and accounts balance.

There are two login portals for GoCardless:
- [The traditional login for the GoCardless services, bank accounts...etc](https://manage.gocardless.com/sign-in) We are not interested in it
- [The open banking platform](https://bankaccountdata.gocardless.com/) You need to create an account here.

Since this error is linked to the refresh token, you do not need an EU banking account to reproduce the bug. **Steps 3 and 4 are optional!**

**Comment by malexanderlim** [comment]
@LucBerge understood, I'm not asking for your credentials, rather, just your email address as support won't look into our ticket or answer any of the questions on our ticket without an active customer. I need to tell support "I'm looking into this issue on behalf of customer [xyz]" otherwise per the automated email that I got back, they won't respond.

I've looked at the ticket you shared and at the OAuth flow and I'm finding the same issue where the refresh token expires every 30 days, and we're not getting a new refresh token in the refresh token request.
